File: Backend/src/main/java/tn/esprit/kaddem/entities/Contrat.java
package tn.esprit.kaddem.entities;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "Contrat")
public class Contrat implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int idContrat;

    boolean archive;

    @Temporal(TemporalType.DATE)
    private Date dateDebutContrat;

    @Temporal(TemporalType.DATE)
    private Date dateFinContrat;

    @Enumerated(EnumType.STRING)
    private Specialité specialite;

    @ManyToOne
    Etudiant etudiant;
}

File: FrontEnd/src/app/core/model/Contrat.ts
import { Etudiant } from './etudiant';
import { Specialite } from './Specialite';
export class Contrat {
  idContrat: number;
  archive: boolean;
  dateDebutContrat: Date;
  dateFinContrat: Date;
  specialite: Specialite;
  etudiant: Etudiant;
  constructor(
    idContrat: number,
    archive: boolean,
    dateDebutContrat: Date,
    dateFinContrat: Date,
    specialite: Specialite,
    etudiant: Etudiant
  ) {
    this.idContrat = idContrat;
    this.archive = archive;
    this.dateDebutContrat = dateDebutContrat;
    this.dateFinContrat = dateFinContrat;
    this.specialite = specialite;
    this.etudiant = etudiant;
  }
}
